scripts/mssql_discover.py

```diff
@@ -0,0 +1,78 @@
+import sqlalchemy
+
+
+# NB. These are connection secrets, never ever commit these
+USER = ""
+PASS = ""
+IP = ""
+PORT = "1433"
+DB = ""
+
+MASTER_MSSQL_URL = f"mssql+pyodbc://{USER}:{PASS}@{IP}:{PORT}/{DB}?driver=ODBC+Driver+17+for+SQL+Server"
+
+
+SUPPORTED_DATA_TYPES = set(
+    [
+        # char types
+        "varchar",
+        "nvarchar",
+        "char",
+        "nchar",
+        "ntext",
+        "text",
+        # numeric types
+        "int",
+        "bigint",
+        "smallint",
+        "tinyint",
+        "money",
+        "float",
+        "decimal",
+        # date types
+        "date",
+        "datetime",
+        "datetime2",
+        "smalldatetime",
+        # other types
+        "bit",
+    ]
+)
+
+
+def mssql_discover():
+    """
+    Select all databases from the instance
+    Select the schema data for each data base
+    Check if there are any fields in the schema that Fidesops does not yet support
+    """
+    engine = sqlalchemy.create_engine(MASTER_MSSQL_URL)
+    all_dbs = engine.execute("SELECT name FROM sys.databases;").all()
+    all_columns = []
+    flagged_columns = []
+    flagged_datatypes = set()
+    for db_name in all_dbs:
+        db_name = db_name[0]
+        try:
+            columns = engine.execute(
+                f"SELECT TABLE_NAME, COLUMN_NAME, DATA_TYPE, IS_NULLABLE FROM {db_name}.INFORMATION_SCHEMA.COLUMNS;"
```

Comment:
I can’t think of any way this this would be set outside this function, but if it was it seems vulnerable to SQL injection.

I don’t see a way to get to it from the outside so I’m probably being over paranoid.